src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -327,7 +327,7 @@ private static Socket Connect(string serverName, int port, TimeSpan timeout, boo
             string IPv4String = null;
             string IPv6String = null;
 
-            Socket[] sockets = new Socket[2];
+            Socket[] sockets = new Socket[ipAddresses.Length];
```

Comment:
You probably also need to use another incrementor inside that double loop, because now it is `sockets[i]` where `i` is `for (int i = 0; i < preferedIPFamilies.Length; ++i)`